src/wallet.rs

```diff
@@ -47,10 +47,10 @@ fn salt_encryption_key(
     Ok(salted_encryption_key)
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub struct Wallet {
     name: String,
-    encryption_key: [u8; 32],
+    encryption_key: Option<([u8; 32])>,
```

Comment:
Do I understand correctly that this is only a temporary measure until we remove the deprecated APIs?

Reply:
yep, I did this so the deprecated API's still work and an error is produced if the deprecated API is used in conjunction with the new API;s